File: C#/BluffinMuffin.Protocol/Lobby/CheckCompatibilityCommand.cs
using BluffinMuffin.Protocol.DataTypes.Attributes;
using BluffinMuffin.Protocol.Enums;

namespace BluffinMuffin.Protocol.Lobby
{
    /// <summary>
    /// This should be sent one time, at the beginning, to know what the server can do.
    /// The supported version, all the supported lobby types and all the supported rules are then sent back to the client.
    /// </summary>
    public class CheckCompatibilityCommand : AbstractLobbyCommand, ICommandWithResponse<CheckCompatibilityResponse>
    {
        /// <summary>
        /// The version of the Implemented Bluffin Protocol by the client, Represented as "Major.Minor.Revision"
        /// </summary>
        [ExampleValue("2.0.0")]
        public string ImplementedProtocolVersion { get; set; }
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public CheckCompatibilityResponse Response(bool success, BluffinMessageId msgId, string message)
        {
            return new CheckCompatibilityResponse(this) { Success = success, MessageId = msgId, Message = message };
        }
    }
}

File: Python/BluffinMuffin.Protocol/lobby/check_compatibility_command.py
from abstract_lobby_command import AbstractLobbyCommand
from data_types.table_params import TableParams

__author__ = 'ericmas001@gmail.com'



class CheckCompatibilityCommand(AbstractLobbyCommand):
    def __init__(self, obj):
        super().__init__(obj)
        self.implemented_protocol_version = obj['ImplementedProtocolVersion']

    def __str__( self ):
        return '{0} ({1})'.format(super().__str__(), self.implemented_protocol_version)
